nixos/modules/services/mail/postfixadmin.nix

```diff
@@ -0,0 +1,198 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.postfixadmin;
+  fpm = config.services.phpfpm.pools.postfixadmin;
+  localDB = cfg.database.host == "localhost";
+  user = if localDB then cfg.database.username else "nginx";
```

Comment:
I think that the naming `database.username` is slightly misleading here. Apart from that, what's wrong with using `DynamicUser` in conjunction with `StateDirectory` (it's getting late, so perhaps I'm missing something :))?

Reply:
Just realized that this wouldn't work since the DynamicUser only exists when the service itself is running, but it has to be started *after* the database initialization is finished which is done in a different service, so this is the correct solution here.

Going to resolve the merge conflict locally and merge then.